cli/src/test/scala-2.12/coursier/cli/util/JsonReportTest.scala

```diff
@@ -0,0 +1,58 @@
+package coursier.cli.util
+
+import coursier.cli.CliTestLib
+import org.junit.runner.RunWith
+import org.scalatest.FlatSpec
+import org.scalatest.junit.JUnitRunner
+
+@RunWith(classOf[JUnitRunner])
+class JsonReportTest extends FlatSpec with CliTestLib {
+  "empty JsonReport" should "be empty" in {
+    val report: String = JsonReport[String](IndexedSeq(), Map())(
+      children = _ => Seq(),
+      reconciledVersionStr = _ => "",
+      requestedVersionStr = _ => "",
+      getFile = _ => Option("")
+    )
+
+    assert(
+      report == "{\"conflict_resolution\":{},\"dependencies\":[],\"version\":\"0.1.0\"}")
```

Comment:
Any particular reason string based assertion is used rather than parsing the json report then assert on structured data? https://github.com/coursier/coursier/blob/master/cli/src/test/scala-2.12/coursier/cli/CliFetchIntegrationTest.scala#L18

Reply:
I think the assertion failure message is a bit clearer with the string, and the string is the actual output, so it feels like the right thing to test against. That said, it could be a bit too fragile.